### ISSUE TITLE
tf.nn.sparse_softmax_cross_entropy_with_logits raise Segmentation fault

### ISSUE THREAD

**Issue by hxsnow10** [tensorflow/tensorflow#11723]
------------------------

### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
code from tensorflow/nmt
- **OS Platform and Distribution**:centos 7
- **TensorFlow installed from**:pip install
- **TensorFlow version**:('v1.2.0-5-g435cdfc', '1.2.1')
- **Python version**: 2.7.5
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:without gpu
- **GPU model and memory**:None
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:
https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

### Describe the problem
As the code pointed, tf.nn.sparse_softmax_cross_entropy_with_logits seems supporting less size tensor than tf.nn.softmax_cross_entropy_with_logits. At least, do shape check rather than Segmenation fault error.

### Source code / logs
```python
import tensorflow as tf
import numpy as np
labels=tf.placeholder(tf.int32, [None,128])
logits=tf.placeholder(tf.float32, [None,128,600000])
crossent1=tf.nn.sparse_softmax_cross_entropy_with_logits(labels=labels, logits=logits)
labels_one_hot=tf.one_hot(labels, 600000, dtype=tf.int32)
crossent2 = tf.nn.softmax_cross_entropy_with_logits(labels=labels_one_hot, logits=logits)
logits_data=np.random.rand(39,128,600000)
labels_data=np.random.randint(600000,size=(39,128))
fd={logits:logits_data, labels:labels_data}
with tf.Session() as sess:
    sess.run([crossent2], feed_dict=fd)
    print 'OK2' # OK2
with tf.Session() as sess:
    sess.run([crossent1], feed_dict=fd)
    print 'OK1' #Segmentation fault
```

**Comment by reedwm** [comment]
I apologize but I am having a hard time understanding what the problem is, where the problem is, and what version it affects. Please resubmit and pay attention to the issue template (https://github.com/tensorflow/tensorflow/issues/new) . Please provide all the information it asks. Thank you.

**Comment by hxsnow10** [comment]
@reedwm i have updated the problem, with my experiment code.

**Comment by reedwm** [comment]
I cannot reproduce. I ran that code after replacing the "600000"s with "6000"s so it would finish in a reasonable amount of time, and the code worked fine. 

[`tf.nn.sparse_softmax_cross_entropy_with_logits` ](https://www.tensorflow.org/api_docs/python/tf/nn/sparse_softmax_cross_entropy_with_logits) requires `labels` to have one less dimension as `logits`. [`tf.nn.softmax_cross_entropy_with_logits`](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits) requires `labels` and `logits` to have the same shape. You do this in your sample code, so I'm not sure what the problem is.

**Comment by hxsnow10** [comment]
@reedwm 6000 work fine, but 600000 not.

**Comment by reedwm** [comment]
With 600000, `logits_data` is 12 GB of data. Using this in a feed dict will then take even more memory. You are probably getting the segmentation fault due to a lack of memory.

**Comment by hxsnow10** [comment]
but tf.softmax works.....
